### PR TITLE
Update promotion

### LINKED ISSUE #13: Ability to Update a Promotion
**As a** Promotions Manager
**I need** a way to update a promotion
**So that** teams can update a promotion by giving the promotion ID

**Details & Assumptions:** 
* The base URL would be `/promotions`
* The HTTP method should be PUT
* The body should contain the details of the updated promotion
* The URL should return the promotion ID of the updated promotion

**Acceptance Criteria:**

```gherkin
Given the server is available
When I call the  `/promotions/{id}` with an HTTP PUT method and the payload as the body
Then it should return the promotion_id of the updated promotion with the HTTP Status Code 200 OK
When an error occurs while hitting the endpoint
Then a proper HTTP Status Code depicting the type of error should be returned
```

**The body should look like this:**

```
{
    name: <name>,
    type: <type>,
    description: <description>,
    promotion_value: <promotion value>,
    promotion_percent: <promotion_percent>,
    status: <promo_status>,
    expiry: <expiry>,
    last_updated_at: <last_updated_at>
}
```

Where <type> can be python enumerations like [ABS_DISCOUNT, PERCENT_DISCOUNT, DELIVER_DISCOUNT, GIFT_CARDS]

**The result should look like this:**

```
{
    id: <promotion_id>
}
```
Where <promotion_id> is the promotion ID of the newly created promotion.

### PR DESCRIPTION
**Pull request by Aman-Chopra**
- Ability to Update a Promotion
- All test cases are Passing
- Code coverage 100%
- No pylint errors

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -47,7 +68,31 @@ def create_promotion():
...
+######################################################################
+# UPDATE AN EXISTING PET
+######################################################################
+
 
+@app.route("/promotions/<int:promotion_id>", methods=["PUT"])
+def update_promotions(promotion_id):
+    """
+    Update a Promotion
+    This endpoint will update a Promotion based the body that is posted
```
Comment (Black-Caesaris) [inline service/routes.py:80]: nit: based on*

**Review by Black-Caesaris — APPROVED** [review]
LGTM!

**Review by Awesalot — APPROVED** [review]
Looks good. I'd suggest commenting in the structure of the promotion in the create function and what parameters we can update in the update function to make it easier to parse and add potential future testcases.

**Comment by Aman-Chopra** [comment]
> Looks good. I'd suggest commenting in the structure of the promotion in the create function and what parameters we can update in the update function to make it easier to parse and add potential future testcases.

Sure! I have added these things in the story. We can also add this in the readme.md

**Pull request merged**